vendor-extra/LiberoPageBundle/src/EventListener/LiberoPageListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\LiberoPageBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+final class LiberoPageListener
+{
+    private $routes;
+
+    /**
+     * @param array<string,string> $routes
+     */
+    public function __construct(array $routes)
+    {
+        $this->routes = $routes;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $route = $request->attributes->get('_route');
+
+        if (!isset($this->routes[$route])) {
+            $request->attributes->set('libero_page', ['type' => '']);
+
+            return;
+        }
+
+        $page = $this->routes[$route];
+
+        if ('content' === $page['type']) {
+            $page['content_id'] = $request->attributes->get('id');
+        }
+
+        $request->attributes->set('libero_page', $page);
```

Comment:
This is the key part. Attaches the details (page type etc) about the Libero page to the request (based on the matched route).